docs/connections.rst

```diff
@@ -0,0 +1,335 @@
+
+      # Ensemble to ensemble
+      nengo.Connection(ens1, ens2)
+      # Ensemble slice to node
+      nengo.Connection(ens1[0], node)
+      # Ensemble to neurons slice
+      nengo.Connection(ens1, ens2.neurons[:2])
+
+The important thing about decoded connections
+is that they do not directly compute the
+``function`` defined for that connection
+(keeping in mind that passing in no function
+is equivalent to passing in the identity function).
+Instead, the function is approximated
+by solving for a set of decoding weights.
+The output of a decoded connection
+is the sum of the pre ensemble's neural activity
+weighted by the decoding weights
+solved for in the build process.
+
+Mathematically, you can think of a decoded connection
+as implementing the following equation:
+
+.. math:: \mathbf{y}(t) = \sum_{i=0}^n \mathbf{d}^{f}_i a_i(x(t))
+
+where
+
+- :math:`\mathbf{y}(t)` is the output of the connection at time :math:`t`,
+- :math:`i` is the number of neurons in the pre ensemble
+- :math:`\mathbf{d}^{f}_i` is the decoding weight associated
+  with neuron :math:`i` given the function :math:`f`,
+- :math:`a_i(x(t))` is the activity of neuron :math:`i` given
+  :math:`x(t)`, the input at time :math:`t`.
+
+Note that the length of the :math:`\mathbf{d}` and :math:`\mathbf{y}` vectors
+is the same, and is specified by the dimensionality of
+the output of the function :math:`f` when applied to input :math:`x`.
+
+While the equation above is straightforward,
+there are several important implications
+that one should keep in mind when using decoded connections.
+
+- Decoders will be automatically solved for in the build process.
+
+  Solving for decoders makes up the majority of build time,
+  so if building your networks takes a long time,
+  look at your decoded connections and
+  try lowering the number of neurons
+  or using different `.Solver` types.
+
+- The function passed to the connection
+  is used to determine decoders.
+  It will never be run during the simulation.
+
+  When you define a function in a node,
+  it will be execute on every simulation timestep.
+  That may lead you to think that the function
+  passed to a connection is executed on every timestep,
+  but that is *not* the case for decoded connections.
+  The function passed to the connection will be executed
+  in the decoder solving process determine an error
+  to minimize, but never during the simulation.
+
+- The characteristics of the ``pre`` ensemble
+  are critically important to performance.
+
+  If you determine that your decoded connection
+  is not approximating the desired function well,
+  examine the ``pre`` ensemble.
+  The decoded value depends on the activity
+  of the ``pre`` ensemble;
+  does it represent its input reliably?
+  If not, then a function of that input
+  cannot be well approximated.
+  If you think that your function may be incorrect,
+  switch the ``pre`` ensemble to use
+  the `.Direct` neuron type,
```

Comment:
Do backticks have any effect in `.rst` file rendering? GitHub doesn't seem to be doing anything with them, but GitHub's rendering of `.rst` files is a bit weird sometimes.